sdk/resourcemanager/Azure.ResourceManager/src/Common/Custom/Models/ManagedServiceIdentity.Serialization.cs

```diff
@@ -18,22 +18,55 @@ namespace Azure.ResourceManager.Models
     [JsonConverter(typeof(ManagedServiceIdentityConverter))]
     public partial class ManagedServiceIdentity : IJsonModel<ManagedServiceIdentity>
     {
-        internal void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options, JsonSerializerOptions jOptions = default)
+        private const string SystemAssignedUserAssignedV3Value = "SystemAssigned,UserAssigned";
+
+        // This method checks if the format string in options.Format ends with the "|v3" suffix.
+        // The "|v3" suffix indicates that the ManagedServiceIdentityType format is version 3.
+        // If the suffix is present, it is removed, and the base format is returned via the 'format' parameter.
+        // This allows the method to handle version-specific logic while preserving the base format.
+        private static bool UseManagedServiceIdentityV3(ModelReaderWriterOptions options, out string format)
         {
-            var format = options.Format == "W" ? ((IPersistableModel<ManagedServiceIdentity>)this).GetFormatFromOptions(options) : options.Format;
+            var originalFormat = options.Format.AsSpan();
+            if (originalFormat.Length > 3)
+            {
+                var v3Format = "|v3".AsSpan();
+                if (originalFormat.Slice(originalFormat.Length - v3Format.Length).Equals(v3Format, StringComparison.Ordinal))
+                {
+                    format = originalFormat.Slice(0, originalFormat.Length - v3Format.Length).ToString();
+                    return true;
+                }
+            }
+
+            format = options.Format;
+            return false;
+        }
+
+        internal void Write(Utf8JsonWriter writer, ModelReaderWriterOptions options, JsonSerializerOptions jOptions = null)
```

Comment:
so after this modification, this `jOptions` parameter is never used in this method now, we could remove it.

Reply:
and once you do this, you would notice the invocation of this method in the converter - it is equivalent to just calling the interface's Write method because we are only passing in the MRWOptions.
Therefore we could remove this method, and put all its implementation into `IJsonModel<T>.Write` method.

Reply:
yes, we should remove this since this is internal, or actually it should be private in the first place.